src/Microsoft.AspNetCore.BuildTools.ApiCheck.Task/ApiCheckTask.cs

```diff
@@ -0,0 +1,188 @@
+            {
+                if (Framework.StartsWith("net4", StringComparison.OrdinalIgnoreCase))
+                {
+                    return ApiCheckToolName + ".exe";
+                }
+
+                var exeExtension = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".exe" : string.Empty;
+                return "dotnet" + exeExtension;
+            }
+        }
+
+        /// <inheritdoc />
+        protected override bool ValidateParameters()
+        {
+            if (string.IsNullOrEmpty(ApiListingPath) || !File.Exists(ApiListingPath))
+            {
+                Log.LogError($"API listing file '{ApiListingPath}' not specified or does not exist.");
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(AssemblyPath) || !File.Exists(AssemblyPath))
+            {
+                Log.LogError($"Assembly '{AssemblyPath}' not specified or does not exist.");
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(Framework))
+            {
+                Log.LogError($"Framework moniker must be specified.");
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(ProjetAssetsPath) || !File.Exists(ProjetAssetsPath))
+            {
+                Log.LogError($"Project assets file '{ProjetAssetsPath}' not specified or does not exist.");
+                return false;
+            }
+
+            if (!string.IsNullOrEmpty(ExclusionsPath) && !File.Exists(ExclusionsPath))
+            {
+                Log.LogError($"Exclusions file '{ExclusionsPath}' does not exist.");
+                return false;
+            }
+
+            return base.ValidateParameters();
+        }
+
+        /// <inheritdoc />
+        protected override string GenerateFullPathToTool()
+        {
+            // ToolExe equals ToolName by default. Assume (as base class does) any user override is in the same directory.
+            if (Framework.StartsWith("net4", StringComparison.OrdinalIgnoreCase))
+            {
+                var taskAssemblyFolder = Path.GetDirectoryName(GetType().GetTypeInfo().Assembly.Location);
+                return Path.GetFullPath(Path.Combine(taskAssemblyFolder, "..", "net452", ToolExe));
+            }
+
+            var dotnetPath = new FileInfo(AppContext.GetData("FX_DEPS_FILE") as string) // .\Microsoft.NETCore.App.deps.json
+                .Directory // version e.g. 1.0.1
+                ?.Parent   // Microsoft.NETCore.App
+                ?.Parent   // shared
+                ?.Parent   // DOTNET_HOME
+                ?.GetFiles(ToolExe)
+                .SingleOrDefault();
+
+            // If above expression does not find dotnet, fall back to system PATH and hope for the best.
+            return dotnetPath?.Exists == true ? dotnetPath.FullName : ToolExe;
+        }
+
+        /// <inheritdoc />
+        protected override string GenerateCommandLineCommands()
+        {
+            var arguments = string.Empty;
```

Comment:
🙈 Not the Right Thing:tm: for this task. `ArgumentEscaper` doesn't handle anything but spaces (not all special `bash` and `cmd` characters) and this will run in /every/ Universe build. In addition, MSBuild properties like `$(TargetPath)` are already in reasonable shape and folder names containing double quotes are incredibly unlikely.